lib/worldwide/region.rb

```diff
@@ -484,6 +485,12 @@ def search_prefixes_by_normalized_zip(prefixes:, zip:, allow_partial_zip: false)
       end&.first
     end
 
+    def iso_code_to_legacy_code(iso_code)
+      return iso_code if iso_code.nil? || iso_code.length < 3
+
+      iso_code.split("-").last
+    end
```

Comment:
🤔 Unfortunately, the situation is more complicated than this.  The legacy codes are a bit of a... legacy... mess.

```ruby
irb(main):001:0> Worldwide.region(code: "MX").zone(code: "DF").iso_code
=> "MX-CMX"
irb(main):002:0> Worldwide.region(code: "MX").zone(code: "DF").legacy_code
=> "DF"
```